GitHub API errors cause user to log out

A GitHub token issue caused this error in the logs:

```text
[2026-04-07T05:10:33.951Z] [ERROR] [github] Failed to fetch PR for branch {"repoFullName":"brendanlong/experiments","branch":"claude/2e59dc9e-d448-4113-bba7-f041b500292e"}
  Error: GitHubApiError: GitHub API error: 401 for /repos/brendanlong/experiments/pulls?head=brendanlong%3Aclaude%2F2e59dc9e-d448-4113-bba7-f041b500292e&state=all&per_page=1&sort=updated&direction=desc
      at P6 (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:105:18726)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async P9 (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:105:18784)
      at async P5 (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:105:19055)
      at async /home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:118:19260
      at async n7.middlewares (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:15:7479)
      at async ir (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:18:69)
      at async ir (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:18:69)
      at async ir (/home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:18:69)
      at async /home/claude/clawed-abode/.next/server/chunks/[root-of-the-server]__38f7d6a0._.js:55:1372
[2026-04-07T05:10:33.952Z] [INFO ] [trpc] query github.getSessionPrStatus {"duration":115}
```

But instead of showing an error message, it redirects me to the login page.

--- a/src/server/routers/github.test.ts
+++ b/src/server/routers/github.test.ts
@@ -519,7 +519,7 @@ describe('githubRouter', () => {
       });
     });
 
-    it('should throw UNAUTHORIZED for invalid token', async () => {
+    it('should throw PRECONDITION_FAILED for invalid token', async () => {
       mockFetch.mockResolvedValue(createMockResponse({}, 401));
 
       const caller = createCaller('auth-session-id');
@@ -530,7 +530,7 @@ describe('githubRouter', () => {
           issueNumber: 123,
         })
       ).rejects.toMatchObject({
-        code: 'UNAUTHORIZED',
+        code: 'PRECONDITION_FAILED',
         message: 'GitHub token is invalid or expired',
       });
     });

--- a/src/server/routers/github.ts
+++ b/src/server/routers/github.ts
@@ -52,7 +52,7 @@ async function githubFetchResponse(path: string, token?: string): Promise<Respon
   if (!response.ok) {
     if (response.status === 401) {
       throw new TRPCError({
-        code: 'UNAUTHORIZED',
+        code: 'PRECONDITION_FAILED',
         message: 'GitHub token is invalid or expired',
       });
     }

